applications/fishing-map/src/features/resources/resources.selectors.ts

```diff
@@ -28,51 +20,7 @@ export const selectVisibleResources = createSelector(
   }
 )
 
-export const getVesselResourceQuery = (
-  dataview: UrlDataviewInstance<Generators.Type>,
-  datasetType: DatasetTypes | DatasetTypes[]
-): Resource | null => {
-  const { url, dataset, datasetConfig } = resolveDataviewDatasetResource(dataview, datasetType)
-  if (url && dataset && datasetConfig) {
-    return {
-      dataviewId: dataview.dataviewId as number,
-      url: url,
-      dataset: dataset,
-      datasetConfig: datasetConfig,
-    }
-  }
-  return null
-}
-
 export const selectDataviewsResourceQueries = createSelector(
-  [selectDataviewInstancesResolved, isGuestUser, selectDebugOptions],
-  (dataviewInstances, guestUser, { thinning }) => {
-    if (!dataviewInstances) return
-    const resourceQueries: Resource[] = dataviewInstances.flatMap((dataview) => {
-      if (dataview.config?.type !== Generators.Type.Track || dataview.deleted) {
-        return []
-      }
-
-      let trackResource: Resource | null = null
-      let eventsResources: (Resource | null)[] = []
-
-      if (dataview.config.visible === true) {
-        const datasetType =
-          dataview.datasets && dataview.datasets?.[0]?.type === DatasetTypes.UserTracks
-            ? DatasetTypes.UserTracks
-            : DatasetTypes.Tracks
-
-        trackResource = getVesselResourceQuery(dataview, datasetType)
-        eventsResources = resolveDataviewEventsResources(dataview)
-      }
-
-      const infoResource = getVesselResourceQuery(dataview, DatasetTypes.Vessels)
-
-      return [trackResource, infoResource, ...eventsResources].filter(
-        (r) => r !== null
-      ) as Resource[]
-    })
-
-    return resourceQueries
-  }
+  [selectDataviewInstancesResolved],
+  (dataviewInstances) => resolveResourceEndpoint(dataviewInstances ?? [])
```

Comment:
I love how clean this is now 👏 but the method name looks confusing, doesn't it?
It receives dataviews and returns queries, so why not calling it something like `resolveDataviewsQueries` ?

Reply:
Thanks! Yes, you are right, I'll rename it to `resolveDataviewsResourceQueries` to express that the queries being resolved belong to resources.